src/main/java/com/autosale/shop/listeners/SessionEventListener.java

```diff
@@ -0,0 +1,20 @@
+package com.autosale.shop.listeners;
+
+import com.autosale.shop.service.TraceHistoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.event.EventListener;
+import org.springframework.data.redis.core.RedisKeyExpiredEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class SessionEventListener {
+
+    private final TraceHistoryService traceHistoryService;
+
+    @EventListener
+    public void onSessionExpireEvent(RedisKeyExpiredEvent<?> event) {
+        traceHistoryService.log(String.format("Session of user %s is expired", new String(event.getId())));
```

Comment:
String format takes obj as a second argument. Don't need to do new String(event.getId()))

Reply:
Yeah, but event.getId returns an array, that is not an object, so anyway we need to cast, somehow, array of bytes to object and I think that casting it directly to string is the best way

Reply:
Aha, I see. It returns an array of bytes. Then good